file-formats/sajc/type/zif_aff_sajc_v1.intf.abap

```diff
@@ -24,6 +24,19 @@ INTERFACE zif_aff_sajc_v1
   "! Name of the program
   TYPES ty_program_name TYPE c LENGTH 40.
 
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! General information
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">Class with Execute Method</p>
+      "! Name of the class which contains the execute-method to run within the job
+      "! $required
```

Comment:
Is`class_name` required? Isn't it sufficient to specify a `program_name`?

Reply:
Some words about the background: There are two different types of catalog entries: class based (which run an ABAP class in background) and report based (which run an ABAP report in background). Because in Steampunk only classes can be created, the ADT only allows to edit class based catalog entries.

Now the problem is that both types use the same transport object, so both types are visible on the left side of the ADT in the tree. This means that it is possible to select a report based catalog entry in the tree. I implemented this in a way that a report based catalog entry uses a different configuration file which sets everything to read-only.

Now there are two problems:
- In a report based catalog entry no class name with title 'Class name' should be displayed, but a report name with title 'Program name'.
- The check method should display an error message which explains that report based catalog entries are not supported by the ADT. So, the check method should recognize that currently a report based catalog entry is displayed.

To realize this I used the following way:
- In a class based catalog entry the field class_name is filled (which is required) and only this field is visible on the screen.
- In a report based catalog entry, the field program_name is filled (which is read-only like everything else) and only this field is visible on the screen. 
- The check method checks whether the field program_name is filled and outputs an error message in this case.

Meanwhile there would be a better solution:
- I think the title of the field can be overwritten in the configuration file, can't it?
- Meanwhile the check method reads the catalog entry from the database (to check whether the user changed the class name and needs to sync the parameter list). So, the check method may also read the information from the database whether the catalog entry is report based. It does not need the field program_name any more,

But I don't think that I can remove the field program_name because of ABAPGit and incompatible change. I don't want to increase the version number because of this.

Reply:
Having both fields is fine for me. I was wondering whether the field `class_name` should be a `$required` field. Because obviously, it is not required, if there is a program.

I would like to double-check with @wurzka.

Reply:
I have opened an issue. See #546.

Maybe, we recommend to increase the format version later to solve the issue. However, I would keep it as it is, for now.